quasar-core/src/main/java/co/paralleluniverse/strands/channels/disruptor/DisruptorReceiveChannel.java

```diff
@@ -42,18 +43,21 @@ public DisruptorReceiveChannel(RingBuffer<Message> buffer, Sequence... dependent
 
     @Override
     public Message receive() throws SuspendExecution, InterruptedException {
-        if (closed)
+        if (closed) {
             return null;
+        }
         long nextSequence = sequence.get() + 1L;
-        while (nextSequence > availableSequence) {
-            try {
+        try {
+            while (nextSequence > availableSequence) {
                 availableSequence = barrier.waitFor1(nextSequence);
-            } catch (AlertException ex) {
-                // ???
             }
+            Message message = buffer.get(nextSequence);
+            sequence.set(nextSequence);
+            return message;
+        } catch (AlertException e) {
+            return null;
```

Comment:
What is `AlertException` used for, and why should we return `null` when it is thrown? We can return `null` from `receive` if and only if the channel is closed. If that exception signals that the channel is closed, we must set the closed status.